include/envoy/network/dns.h

```diff
@@ -39,7 +39,8 @@ class DnsResolver {
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name, uint32_t port,
```

Comment:
@mattklein123 

Currently, the `Host` interface requires weights to be in the range 1-100. I'm not sure how to handle out of range weights in the case of logical DNS clusters. Maybe more SRV config options will be needed. So I think the config may look like this,

```json
"hosts": [{"url": "tcp://foo.service"}],
"srv_config": {
   "some_option": "some_value"
}
```

Then
+ Change `Utility::resolveUrl` `Utility::hostFromTcpUrl` and `Utility::portFromTcpUrl` to handle this case.
+ Add a wrapper type for an `Address::Instance` with priority and weight. Though I'm not sure about the name.
+ Change the `ResolveCb` type accordingly.
+ Add a `resolveSrv` method.
+ Use this new address type to construct `HostImpl`s.

What do you think?

PS: I might not be able to continue working on this as the current implementation already met our needs. Maybe someone can start from here.